src/Microsoft.XmlSerializer.Generator/pkg/build/Microsoft.XmlSerializer.Generator.targets

```diff
@@ -17,11 +17,48 @@
     <Warning Condition="Exists('$(SerializerDllImmediatePath)') != 'true' And Exists('$(SerializerCsImmediatePath)') == 'true'" Text="$(SGenWarningText)"/>
     <Copy Condition="Exists('$(SerializerDllImmediatePath)') == 'true'" SourceFiles="$(SerializerDllImmediatePath)" DestinationFolder="$(OutputPath)" />
   </Target>
+  
   <Target Name="CleanSerializationAssembly" AfterTargets="CoreClean">
     <Message Text="Cleaning serialization files..." Importance="normal"/>
     <Delete Condition="Exists('$(OutputPath)\$(SerializerName).dll') == 'true'" Files="$(OutputPath)\$(SerializerName).dll" />
   </Target>
+  
   <Target Name="CopySerializer" AfterTargets="PrepareForPublish">
     <Copy Condition="Exists('$(OutputPath)\$(AssemblyName).XmlSerializers.dll')=='true'" SourceFiles="$(OutputPath)\$(AssemblyName).XmlSerializers.dll" DestinationFolder="$(PublishDir)" SkipUnchangedFiles="false" />
   </Target>
+  
+ <Target Name="GenerateSerializationAssemblyForReferenceAssembly" AfterTargets="GenerateSerializationAssembly" Condition="@(SerializationAssembly)!=''">
+    <ItemGroup Condition="@(SerializationAssembly)!=''" Label="Parse SerializationAssembly">
+      <SearchSerializationAssembly Include="@(Reference)">
+        <SerializationAssembly>%(SerializationAssembly.Identity)</SerializationAssembly>
+      </SearchSerializationAssembly>
+      <TargetSerializationAssembly Include="@(SearchSerializationAssembly)" Condition="$([System.String]::new('%(SearchSerializationAssembly.Identity)').EndsWith('%(SearchSerializationAssembly.SerializationAssembly).dll'))" />
+      <TargetSerializationAssemblyAndType Include="@(TargetSerializationAssembly)">
+        <TargetSerializationAssembly>%(SerializationAssembly.Identity)</TargetSerializationAssembly>
+        <TargetSerializationType>@(SerializationAssembly-&gt;Metadata('SerializationType'))</TargetSerializationType>
+      </TargetSerializationAssemblyAndType>
+    </ItemGroup>
```

Comment:
The `ItemGroup` above can be simplified as,

```xml
    <ItemGroup Condition="@(SerializationAssembly)!=''" Label="Parse SerializationAssembly">
      <SearchSerializationAssembly Include="@(Reference)">
        <AssemblyName>%(SerializationAssembly.Identity)</AssemblyName>
        <SerializationTypes>%(SerializationAssembly.SerializationType)</SerializationTypes>
      </SearchSerializationAssembly>
      <TargetSerializationAssembly Include="@(SearchSerializationAssembly)" Condition="$([System.String]::new('%(SearchSerializationAssembly.Identity)').EndsWith('%(SearchSerializationAssembly.AssemblyName).dll'))" />
    </ItemGroup>
```

Accordingly, the `<Exec>` at line 53 can be simplified as,

```xml
<Exec Command="dotnet Microsoft.XmlSerializer.Generator /force /quiet /assembly:%(TargetSerializationAssembly.Identity) /type:%(TargetSerializationAssembly.SerializationTypes) /out:$(IntermediateOutputPath)" ContinueOnError="true" />
```